src/Xamarin.Android.Build.Tasks/Tasks/AndroidApkSigner.cs

```diff
@@ -138,6 +138,14 @@ protected override bool ValidateParameters ()
 					Log.LogCodedError ("XA4310", Properties.Resources.XA4310, "$(AndroidSigningKeyStore)", KeyStore);
 					return false;
 				}
+				if (string.IsNullOrEmpty (KeyPass)) {
+					Log.LogCodedError ("XA4314", Properties.Resources.XA4310, "$(AndroidSigningKeyPass)");
+					return false;
+				}
+				if (string.IsNullOrEmpty (StorePass)) {
+					Log.LogCodedError ("XA4314", Properties.Resources.XA4310, "$(AndroidSigningStorePass)");
+					return false;
+				}
```

Comment:
I think these have the wrong message: `Properties.Resources.XA4310`, and the code is `XA4314`.

Reply:
🤦

Reply:
Actually , no. I added a new code XA4314 for this specific error. The XA4310 is for files existing, in this case Xa4314 is for properties being empty or null.

Reply:
Maybe just add a comment, if we see this later. Someone might try to change it to make the numbers match.

Reply:
I just realised what you were talking about lol. I updated the `Properties.Resources.XA4310` to be `Properties.Resources.XA4314` as it should have been. It was a place holder until I compiled the new `resx` lol.